docs/docs/pii-management.mdx

```diff
@@ -0,0 +1,259 @@
+anonymizing data in flight could have undesirable consequences for the assistant's dialogue management action predictions.
+
+## Architecture Overview
+
+The anonymization of the Rasa event is run through an anonymization pipeline at the end of the dialogue management
+action prediction and execution. Note that the processing done by the anonymization pipeline is scheduled as a background
+task and does not affect the assistant's response time.
+
+The anonymization steps are as follows:
+1. The Rasa agent will invoke the anonymization pipeline during each user message handling.
+2. The anonymization pipeline will run through a sequence of anonymization rules that are applied to the new events.
+3. The pipeline will publish the anonymized events to the Kafka topic that is mapped to the anonymization rule list in
+`endpoints.yml`.
+4. The tracker store saves the original events which are not anonymized.
+
+### Supported Rasa Events
+
+The Rasa events that are anonymized include the following:
+- `user`
+- `bot`
+- `slot`
+- `entities`
+
+### Supported PII entity types
+
+The anonymization pipeline uses [Microsoft Presidio](https://microsoft.github.io/presidio/text_anonymization/) as both entity recognizer and
+anonymizer. Presidio is an open-source library that supports a wide range of entity types and anonymization methods.
+
+You can specify any of the out-of-the-box [supported Presidio entity types](https://microsoft.github.io/presidio/supported_entities/) in the anonymization rules.
+Note that it is currently not possible to add custom entity types for the Rasa Pro anonymization pipeline.
+
+## How to write anonymization rules
+
+You can now write anonymization rules in `endpoints.yml` to explicitly declare which Presidio entities should be anonymized.
+The anonymization pipeline is configurable via the `anonymization` section in the `endpoints.yml` file.
+This section must have the structure given in the following example:
+
+```yaml
+anonymization:
+  metadata:
+    language: en
+    model_name: en_core_web_lg
+    model_provider: spacy
+  rule_lists:
+    - id: rules_1
+      rules:
+        - entity: PERSON
+          substitution: text
+          value: John Doe
+        - entity: LOCATION
+          substitution: faker
+        - entity: CREDIT_CARD
+          substitution: faker
+        - entity: IBAN_CODE
+          substitution: faker
+    - id: rules_2
+      rules:
+        - entity: CREDIT_CARD
+          substitution: mask
+        - entity: IBAN_CODE
+          substitution: mask
+```
+
+### How to populate the metadata section
+
+The metadata section contains the following fields: `language`, `model_name`, and `model_provider`.
+
+The `language` field specifies the language code of the text to be anonymized. Note that you can only specify one language
+per anonymization pipeline because the assumption made was that an assistant will be developed for a single language.
```

Comment:
```suggestion
per anonymization pipeline.
```

Reply:
I deleted this because it made us look like we are building explicitly for single language bots. But perhaps if we phrased it differently?

Reply:
We need to document this assumption because the pipeline won't be able to work with language switching at the moment. How about: `because the capability is currently not able to handle language switching by end users.`

Reply:
Or maybe replace `because` with `therefore` 😄